### PR TITLE
Cleanup API_Exception usage

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Fixes mis-capitalized exception class name, and moves a v3-only check to the right place.

### CODE REVIEW

File: Civi/API/Kernel.php [hunk 1]
```diff
@@ -157,17 +157,16 @@ public function runRequest($apiRequest) {
...
     require_once 'api/Exception.php';
-
-    if (!is_array($apiRequest['params'])) {
-      throw new \API_Exception('Input variable `params` is not an array', 2000);
-    }
     switch ($apiRequest['version']) {
       case 3:
+        if (!is_array($apiRequest['params'])) {
+          throw new \API_Exception('Input variable `params` is not an array', 2000);
+        }
```
Comment (colemanw) [inline Civi/API/Kernel.php]: @eileenmcnaughton this is one performance thing to note: every time you try to access an api `Action` class params as an array it forces the object to serialize its properties. Here it was being done for no reason.

Comment (eileenmcnaughton) [reply Civi/API/Kernel.php]: Yeah - we no longer need that check IMHO because I think the civicrm_api uses strict type casing

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by colemanw** [comment]
test this please

**Pull request merged**